py/desispec/preproc.py

```diff
@@ -565,9 +565,16 @@ def preproc(rawimage, header, primary_header, bias=True, dark=True, pixflat=True
         for j in range(nrows) :
             readnoise[kk][j] = rdnoise[j]
 
-        header['OVERSCN'+amp] = median_overscan
-        header['OBSRDN'+amp] = median_rdnoise
-
+        header['OVERSCN'+amp] = (median_overscan,'ADUs (gain not applied)')
+        if gain != 1 :
+            rdnoise_message = 'electrons (gain is applied)'
+            gain_message    = 'e/ADU (gain applied to image)'
+        else :
+            rdnoise_message = 'ADUs (gain not applied)'
+            gain_message    = 'gain not applied to image'
+        header['OBSRDN'+amp] = (median_rdnoise,rdnoise_message)
+        header['GAIN'+amp] = (gain,gain_message)
```

Comment:
This trick didn't work.  You are ending up with files with gains like "(1.0, 'gain not applied to image')" rather than a gain of 1.0 with that comment.  That is causing the Travis test failures like "TypeError: Cannot cast ufunc multiply output from dtype('<U32') to dtype('float64') with casting rule 'same_kind'" which should be reproducible by running `python py/desispec/test/test_preproc.py` on your laptop.

Reply:
It works fine when preprocessing images. The header are correct, with values and comments.

The issue was in the test itself that runs several times preproc(..., self.header, ....)
Gains tuples (value,message) are set at the first run, then used in the second and that's where the code fails because it expects a scalar and not a tuple.

I fixed that by forcing a copy of the input header the preproc routine.